### PR TITLE
Resprites Medical Tools And Scanners

### PR DESCRIPTION
**Pull request by Zevotech**
## About The Pull Request
Resprites the basic and advanced surgery tools, as well as adding proper inhands for advanced tools.
Also resprites the health analyzer, advanced health analyzer and reagent scanner.
ALSO makes the health analyzer and advanced health analyzer change sprites and inhands upon changing scan modes.
exam3
exam
exam1
**Sprites done by Any%. God Bless.**

## Why It's Good For The Game
Gives basic surgery tools and the health analyzer a nicer look, while making the advanced tools look cool as shit. Simple as.

## Changelog
:cl:
add: More and better surgical tool sprites and inhands.
code: Health analyzers actually change sprites based on scanmode.
/:cl:
please just let me be done.

### CODE REVIEW

**Comment by zestybastard** [comment]
thanks for all the custom code 👍

**Review by triplezeta — CHANGES_REQUESTED** [review]
Looks great, but really needs a style pass.

**Comment by Ms-Mee** [comment]
Someone who should be in the changelog isn't there, how sus

**Comment by Zevotech** [comment]
> Looks great, but really needs a style pass.

image

**Comment by zestybastard** [comment]
> Looks great, but really needs a style pass.

TetraZeta The Missile. The Missile TetraZetra. You Have To Watch Out

**Comment by Ms-Mee** [comment]
> Someone who should be in the changelog isn't there, how sus

(They are too distracted by zeter and cannot see me if I stand perfectly still, because I have chameleon)

**Comment by Sun-Soaked** [comment]
> Looks great, but really needs a style pass.
 
 I honestly feel like it's the rest of our sprites that need a style pass to match these ones
 Imagine a bright future of pastel 80's techwave for all of our stuff ingame

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -77,36 +77,52 @@ GENE SCANNER
...
+    throw_range = 7
+    custom_materials = list(/datum/material/iron=200)
+    var/mode = SCANNER_VERBOSE
+    var/scanmode = SCANMODE_HEALTH
+    var/advanced = FALSE
+    var/healthscan = "analyzer-1"
+    var/reagentscan = "reagentanalyzer"
+    var/healthinhand = "analyzer"
+    var/reagentinhand = "reagentanalyzer-1"
+    custom_price = 300
```
Comment (MarkSuckerberg) [inline code/game/objects/items/devices/scanners.dm]: Should be tabs instead of spaces

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -403,10 +419,17 @@ GENE SCANNER
...
+    icon_state = "advanalyzer"
+    item_state = "advanalyzer"
+    lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+    righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
+    desc = "A hand-held body scanner able to distinguish vital signs of the subject with high accuracy."
+    advanced = TRUE
+    healthscan = "advanalyzer"
+    reagentscan = "advreagentanalyzer"
+    healthinhand = "advanalyzer"
+    reagentinhand = "advreagentanalyzer"
```
Comment (MarkSuckerberg) [inline code/game/objects/items/devices/scanners.dm]: Tabs

File: code/modules/surgery/tools.dm [hunk 3]
```diff
@@ -282,37 +281,36 @@
...
-		icon_state = "surgicaldrill_a"
+    playsound(get_turf(user), 'sound/weapons/tap.ogg', 50, TRUE)
+    if(tool_behaviour == TOOL_DRILL)
+        tool_behaviour = TOOL_CAUTERY
+        to_chat(user, "<span class='notice'>You focus the lenses of [src], it is now in mending mode.</span>")
+        icon_state = "advcautery"
+    else
+        tool_behaviour = TOOL_DRILL
+        to_chat(user, "<span class='notice'>You dilate the lenses of [src], it is now in drilling mode.</span>")
+        icon_state = "advdrill"
```
Comment (MarkSuckerberg) [inline code/modules/surgery/tools.dm]: Tabs

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

**Comment by Zevotech** [comment]
fixed the tabs, i actually kept using them out of muscle memory and kept getting inconsistent indentation errors so i just converted them to spaces

**Comment by Zevotech** [comment]
(by that i mean theyre all tabs now, i just changed them to spaces while working on the pr cause i didnt want to fuck with the indentation and cause some kind of error down the line)

**Comment by Zevotech** [comment]
This Unpleasant Squiggle Shows Up At Your Door
image

**Comment by Zevotech** [comment]
Im Going To Cry

**Comment by Sylphily** [comment]
> Im Going To Cry

oof, that seems rough. gl

**Comment by Zevotech** [comment]
now just to beg for mergies before the stale label murders my hopes and dreams

**Comment by zestybastard** [comment]
> now just to beg for mergies before the stale label murders my hopes and dreams

dont hold your breath, the sprites need more outline so i have to go add that soon

**Comment by Zevotech** [comment]
hey can we testmerge this on tomorrow's uptime?

**Comment by rye-rice** [comment]
were the sprites outlinfified?

**Comment by Zevotech** [comment]
> were the sprites outlinfified?

image

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -95,18 +95,34 @@ GENE SCANNER
...
+		icon_state = reagentscan
+		item_state = reagentinhand
+		lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+		righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
+	else
+		to_chat(user, "<span class='notice'>You switch the health analyzer to check physical health.</span>")
+		icon_state = healthscan
+		item_state = healthinhand
+		lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+		righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
```
Comment (Fikou) [inline code/game/objects/items/devices/scanners.dm:122]: use ternary operators i.e.
variable = scan_mode ? "thing1" : "thing2"

File: code/game/objects/items/devices/scanners.dm [hunk 5]
```diff
@@ -95,18 +95,34 @@ GENE SCANNER
...
 /obj/item/healthanalyzer/attack_self(mob/user)
+	playsound(get_turf(user), 'sound/machines/click.ogg', 50, TRUE)
 	scanmode = !scanmode
-	to_chat(user, "<span class='notice'>You switch the health analyzer to [scanmode ? "scan chemical contents" : "check physical health"].</span>")
+	if(scanmode == 1)
+		to_chat(user, "<span class='notice'>You switch the health analyzer to scan chemical contents.</span>")
+		icon_state = reagentscan
+		item_state = reagentinhand
+		lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+		righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
```
Comment (Fikou) [inline code/game/objects/items/devices/scanners.dm:116]: why are you setting these

File: code/game/objects/items/devices/scanners.dm [hunk 6]
```diff
@@ -95,18 +95,34 @@ GENE SCANNER
 	var/mode = SCANNER_VERBOSE
 	var/scanmode = SCANMODE_HEALTH
 	var/advanced = FALSE
+	var/healthscan = "analyzer-1"
+	var/reagentscan = "reagentanalyzer"
+	var/healthinhand = "analyzer"
+	var/reagentinhand = "reagentanalyzer-1"
```
Comment (Fikou) [inline code/game/objects/items/devices/scanners.dm:101]: these variables need better names

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by Vexylius** [comment]
@Zevotech hi

**Comment by Zevotech** [comment]
> @Zevotech hi

I'll save changes for when any% gives me the new sprites for these. you know the ones with the outlines.

oh. 
image

**Comment by Sylphily** [comment]
bruh

**Comment by Zevotech** [comment]
we live in a spriteciety

**Comment by Zevotech** [comment]
image0-2

**Pull request closed without merging**